packages/react-vapor/src/components/table-hoc/TableSelectors.ts

```diff
@@ -19,6 +20,19 @@ const initialTableSort: ITableWithSortState = {
 const getIsEmpty = (state: IReactVaporState, props: TableSelectorsProps): boolean =>
     props.data !== null && (!props.data || props.data.length === 0);
 
+const getIsTruelyEmpty = (state: IReactVaporState, props: TableSelectorsProps): boolean => {
+    const compositeState = TableHOCUtils.getCompositeState(props.id, state);
+    const isEmpty = getIsEmpty(state, props);
+
+    const hasNoAppliedFilter = !compositeState.filter;
+    const hasNoAppliedPredicates = !compositeState.predicates?.length;
+    const hasNoAppliedDateLimits =
+        !compositeState.dateLimits?.length ||
+        (compositeState.dateLimits?.[0] == null && compositeState.dateLimits?.[1] == null);
+
+    return isEmpty && hasNoAppliedFilter && hasNoAppliedDateLimits && hasNoAppliedPredicates;

```

Comment:
Not sure if it's worth it but since you use `getIsEmpty` but could you use it as a conditional, so the other checks are only run if `isEmpty` is true?